src/components/InngestFunction.ts

```diff
@@ -68,16 +68,17 @@ export class InngestFunction<Events extends Record<string, EventPayload>> {
      * This function can't be expected to know how it will be accessed, so
      * relies on an outside method providing context.
      */
-    baseUrl: URL
+    baseUrl: URL,
+    appPrefix?: string
   ): FunctionConfig {
     return {
-      id: this.id,
+      id: this.id(appPrefix),
```

Comment:
Nit: Can we do this once to avoid an accidental mismatch of IDs? I made this mistake a while ago when I changed the top `id` but not the ID in the step URL 😆 

```ts
const id = this.id(appPrefix);

return {
  id,
  ...
```